gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsBatchTest.java

```diff
@@ -0,0 +1,649 @@
+    Page<Zone> page = batchResult.get();
+    assertEquals(PAGE_TOKEN, page.nextPageCursor());
+    Iterator<Zone> iterator = page.values().iterator();
+    int resultSize = 0;
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock).times(ZONE_LIST.size() + 1);
+    EasyMock.replay(dns);
+    Zone zoneInfoFunctional = new Zone(dns, new ZoneInfo.BuilderImpl(ZONE_INFO));
+    while (iterator.hasNext()) {
+      assertEquals(zoneInfoFunctional, iterator.next());
+      resultSize++;
+    }
+    assertEquals(ZONE_LIST.size(), resultSize);
+  }
+
+  @Test
+  public void testCreateZone() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<ManagedZone>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    Capture<ManagedZone> capturedZone = Capture.newInstance();
+    batchMock.addCreateZone(EasyMock.capture(capturedZone), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Zone> batchResult = dnsBatch.createZone(ZONE_INFO);
+    assertEquals(0, capturedOptions.getValue().size());
+    assertEquals(ZONE_INFO.toPb(), capturedZone.getValue());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<ManagedZone> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a DnsException on error.");
+    } catch (DnsException ex) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testCreateZoneWithOptions() {
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock);
+    EasyMock.replay(dns);
+    EasyMock.reset(batchMock);
+    EasyMock.reset(optionsMock);
+    EasyMock.expect(optionsMock.service()).andReturn(dns);
+    EasyMock.replay(optionsMock);
+    Capture<RpcBatch.Callback<ManagedZone>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    Capture<ManagedZone> capturedZone = Capture.newInstance();
+    batchMock.addCreateZone(EasyMock.capture(capturedZone), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Zone> batchResult = dnsBatch.createZone(ZONE_INFO, ZONE_FIELDS);
+    assertEquals(ZONE_INFO.toPb(), capturedZone.getValue());
+    assertNotNull(callback.getValue());
+    String selector = (String) capturedOptions.getValue().get(ZONE_FIELDS.rpcOption());
+    assertTrue(selector.contains(Dns.ZoneField.CREATION_TIME.selector()));
+    assertTrue(selector.contains(Dns.ZoneField.NAME.selector()));
```

Comment:
I do not think it is worth it given the effort needed. It requires calculating the length of each selector (accounting for the syntax) and would need to be adjusted if options in the method call change.